src/bridges/notional/NotionalBridge.sol

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Aztec.
+pragma solidity >=0.8.4;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {AztecTypes} from "../../aztec/libraries/AztecTypes.sol";
+import {ErrorLib} from "../base/ErrorLib.sol";
+import {BridgeBase} from "../base/BridgeBase.sol";
+import {IWrappedfCashFactory} from "../../interfaces/notional/IWrappedfCashFactory.sol";
+import {IWrappedfCash} from "../../interfaces/notional/IWrappedfCash.sol";
+import {IRollupProcessor} from "../../aztec/interfaces/IRollupProcessor.sol";
+import {WETH9} from "../../interfaces/notional/WETH9.sol";
+import {CToken} from "../../interfaces/notional/CToken.sol";
+import {NotionalViews, Token} from "../../interfaces/notional/INotionalViews.sol";
+
+contract NotionalBridgeContract is BridgeBase {
+    IWrappedfCashFactory public constant FCASH_FACTORY =
+        IWrappedfCashFactory(0x5D051DeB5db151C2172dCdCCD42e6A2953E27261);
+    NotionalViews public constant NOTIONAL_VIEW = NotionalViews(0x1344A36A1B56144C3Bc62E7757377D288fDE0369);
+    address public constant ETH = address(0);
+    address public constant WETH = address(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    mapping(address => mapping(address => bool)) public preapproved;
+
+    constructor(address _rollupProcessor) BridgeBase(_rollupProcessor) {
+        uint16 maxCurrencyId = NOTIONAL_VIEW.getMaxCurrencyId();
+        for (uint16 i = 1; i <= maxCurrencyId; ) {
+            (Token memory assetToken, Token memory underlyingToken) = NOTIONAL_VIEW.getCurrency(i);
+            ERC20(assetToken.tokenAddress).approve(_rollupProcessor, type(uint256).max);
+            preapproved[assetToken.tokenAddress][_rollupProcessor] = true;
+            if (underlyingToken.tokenAddress != ETH) {
+                ERC20(underlyingToken.tokenAddress).approve(_rollupProcessor, type(uint256).max);
+                preapproved[underlyingToken.tokenAddress][_rollupProcessor] = true;
+            }
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    receive() external payable {}
+
+    /**
+     * @notice Convert function called by rollup processor, will enter or exit Notional lending position
+     * @dev Will revert if the input asset or the output asset is invalid
+     * @param _inputAssetA The input asset. For entering the market, we support ETH, USDC, DAI, WBTC
+     * and their corresponding ctokens. For exiting the market, we support every valid fcash token.
+     * @param _outputAssetA The output asset. For entering the market, it should be a valid fcash token.
+     For exiting the market, this should be one of ETH, USDC, DAI, WBTC, and their corresponding ctokesn
```

Comment:
Please generalise this comment to not list the underlying exactly - Notional could add market.